daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/ExpressionEvaluatingUnparsers.scala

```diff
@@ -103,13 +102,22 @@ class NewVariableInstanceStartUnparser(override val context: VariableRuntimeData
   override lazy val childProcessors = Vector()
 
   override def unparse(state: UState) = {
-    state.variableMap.newVariableInstance(context)
+    val v = state.variableMap.newVariableInstance(context)
+
     if (context.maybeDefaultValueExpr.isDefined) {
-      val maybeVar = state.variableMap.find(context.globalQName)
-      Assert.invariant(maybeVar.isDefined)
-      maybeVar.get.setState(VariableInProcess)
-      val suspendableExpression = new NewVariableInstanceSuspendableExpression(context.maybeDefaultValueExpr.get, context)
-      suspendableExpression.run(state)
+      val dve = context.maybeDefaultValueExpr.get
+      dve match {
+        case constExpr: ConstantExpression[_] => // Do nothing, value has already been computed at schema compile time
```

Comment:
Not sure I understand this. If the defaultValue for the NVI is constant, we still need to get that constant value and set it as the defalutValue, right? Seems like this should be something like
```scala
v.setDefalutValue(constExpr.evaluate(state))
```
Though, this is essentially what the NewVariableInstanceSuspendableExpression will do, so I'm not sure if we need a special case for constant. I guess it does avoid allocating this suspendable thing, but that's probably not too big of a deal? And since the expression is constant, it shouldn't even suspend, so we will avoid that overhead regardless.

Can we remove this special case and just always do the below?